File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
         int missingNumber = 1;
         Arrays.sort(nums);
         Set<Integer> numSet = new HashSet<>();

        for(int number : nums){
            if(number > 0){
                numSet.add(number);
                 while(numSet.contains(missingNumber))
                 missingNumber++;
            }
        }
        return missingNumber;
    }
};

File: 0041-first-missing-positive/0041-first-missing-positive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 function firstMissingPositive (nums) {
    nums.sort((a,b) => a - b);
    let nonRepeat = new Set(nums);
    nums = [...nonRepeat];
    let missingNumber = 1;
    for(let i = 0; i < nums.length; i++){
        if(nums[i] > 0){
            if(nums[i] == missingNumber){
                missingNumber++;
            }else{
                break;
            }
        }
    }   
    return missingNumber;
};